frb_codegen/src/generator/dart/mod.rs

```diff
@@ -403,7 +408,11 @@ fn generate_api_func(func: &IrFunc, ir_file: &IrFile) -> GeneratedApiFunc {
                 "".to_string()
             }
         })
+        // If struct has a method with first element `input0`
         || (input_0.is_some() && MethodNamingUtil::struct_has_methods(ir_file, input_0.unwrap()))
+        //If output is a struct with methods
+        || (func_output_struct_name.is_some()
+            && MethodNamingUtil::has_methods(&func_output_struct_name.unwrap(), ir_file))
```

Comment:
nit: Shall we merge `has_methods` and `struct_has_methods` functions? They look a bit similar. (method_utils.rs)

Reply:
They use different techniques. One does a look on the name and also searches using the method name, the other searches for the first argument receiving a self to the struct type

Reply:
I see. That is partially why I think it may be a bit confusing...

Reply:
removed!